### ISSUE TITLE
Setup failed (npm start and npm test), suspect react-router

### ISSUE THREAD

**Issue by tophsic** [auth0-community/auth0-react-scripts#7]
### Description

I just wanted to try a React project with authentication with auth0-react-scripts.

### Expected behavior

Running Setup instructions, I expect to be able to have a running app and a successfull tests suite.

### Actual behavior

I got errors on `npm start` and `npm test` commands.

#### `npm start`

```bash
> npm start
```

It opens a Chrome tab but I got errors on console:

```
warning.js:36 Warning: Failed prop type: The prop `history` is marked as required in `Router`, but its value is `undefined`.
    in Router (at App.js:12)
    in App (at index.js:7)
printWarning @ warning.js:36
warning @ warning.js:60
checkReactTypeSpec @ checkReactTypeSpec.js:80
validatePropTypes @ ReactElementValidator.js:151
createElement @ ReactElementValidator.js:194
render @ App.js:54
(anonymous) @ ReactCompositeComponent.js:796
measureLifeCyclePerf @ ReactCompositeComponent.js:75
_renderValidatedComponentWithoutOwnerOrContext @ ReactCompositeComponent.js:795
_renderValidatedComponent @ ReactCompositeComponent.js:822
performInitialMount @ ReactCompositeComponent.js:362
mountComponent @ ReactCompositeComponent.js:258
mountComponent @ ReactReconciler.js:46
performInitialMount @ ReactCompositeComponent.js:371
mountComponent @ ReactCompositeComponent.js:258
mountComponent @ ReactReconciler.js:46
mountComponentIntoNode @ ReactMount.js:104
perform @ Transaction.js:140
batchedMountComponentIntoNode @ ReactMount.js:126
perform @ Transaction.js:140
batchedUpdates @ ReactDefaultBatchingStrategy.js:62
batchedUpdates @ ReactUpdates.js:97
_renderNewRootComponent @ ReactMount.js:320
_renderSubtreeIntoContainer @ ReactMount.js:401
render @ ReactMount.js:422
(anonymous) @ index.js:21
(anonymous) @ bundle.js:1372
__webpack_require__ @ bundle.js:556
fn @ bundle.js:87
(anonymous) @ multi_main:3
(anonymous) @ bundle.js:589
__webpack_require__ @ bundle.js:556
(anonymous) @ bundle.js:579
(anonymous) @ bundle.js:582
Router.js:43 Uncaught TypeError: Cannot read property 'location' of undefined
    at new Router (Router.js:43)
    at eval (ReactCompositeComponent.js:295)
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)
    at ReactCompositeComponentWrapper._constructComponentWithoutOwner (ReactCompositeComponent.js:294)
    at ReactCompositeComponentWrapper._constructComponent (ReactCompositeComponent.js:280)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:188)
    at Object.mountComponent (ReactReconciler.js:46)
    at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:371)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:258)
    at Object.mountComponent (ReactReconciler.js:46)
Router @ Router.js:43
(anonymous) @ ReactCompositeComponent.js:295
measureLifeCyclePerf @ ReactCompositeComponent.js:75
_constructComponentWithoutOwner @ ReactCompositeComponent.js:294
_constructComponent @ ReactCompositeComponent.js:280
mountComponent @ ReactCompositeComponent.js:188
mountComponent @ ReactReconciler.js:46
performInitialMount @ ReactCompositeComponent.js:371
mountComponent @ ReactCompositeComponent.js:258
mountComponent @ ReactReconciler.js:46
performInitialMount @ ReactCompositeComponent.js:371
mountComponent @ ReactCompositeComponent.js:258
mountComponent @ ReactReconciler.js:46
mountComponentIntoNode @ ReactMount.js:104
perform @ Transaction.js:140
batchedMountComponentIntoNode @ ReactMount.js:126
perform @ Transaction.js:140
batchedUpdates @ ReactDefaultBatchingStrategy.js:62
batchedUpdates @ ReactUpdates.js:97
_renderNewRootComponent @ ReactMount.js:320
_renderSubtreeIntoContainer @ ReactMount.js:401
render @ ReactMount.js:422
(anonymous) @ index.js:21
(anonymous) @ bundle.js:1372
__webpack_require__ @ bundle.js:556
fn @ bundle.js:87
(anonymous) @ multi_main:3
(anonymous) @ bundle.js:589
__webpack_require__ @ bundle.js:556
(anonymous) @ bundle.js:579
(anonymous) @ bundle.js:582
```

#### `npm test`

```bash
> npm test
 FAIL  src/components/App.test.js
  ● renders without crashing

    TypeError: Cannot read property 'location' of undefined
      
      at new Router (node_modules/react-router/Router.js:43:52)
      at node_modules/react-dom/lib/ReactCompositeComponent.js:295:18
      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:75:12)
      at ReactCompositeComponentWrapper._constructComponentWithoutOwner (node_modules/react-dom/lib/ReactCompositeComponent.js:294:16)
      at ReactCompositeComponentWrapper._constructComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:280:21)
      at ReactCompositeComponentWrapper.mountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:188:21)
      at Object.mountComponent (node_modules/react-dom/lib/ReactReconciler.js:46:35)
      at ReactCompositeComponentWrapper.performInitialMount (node_modules/react-dom/lib/ReactCompositeComponent.js:371:34)
      at ReactCompositeComponentWrapper.mountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:258:21)
      at Object.mountComponent (node_modules/react-dom/lib/ReactReconciler.js:46:35)
      at ReactCompositeComponentWrapper.performInitialMount (node_modules/react-dom/lib/ReactCompositeComponent.js:371:34)
      at ReactCompositeComponentWrapper.mountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:258:21)
      at Object.mountComponent (node_modules/react-dom/lib/ReactReconciler.js:46:35)
      at mountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:104:32)
      at ReactReconcileTransaction.perform (node_modules/react-dom/lib/Transaction.js:140:20)
      at batchedMountComponentIntoNode (node_modules/react-dom/lib/ReactMount.js:126:15)
      at ReactDefaultBatchingStrategyTransaction.perform (node_modules/react-dom/lib/Transaction.js:140:20)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactDefaultBatchingStrategy.js:62:26)
      at Object.batchedUpdates (node_modules/react-dom/lib/ReactUpdates.js:97:27)
      at Object._renderNewRootComponent (node_modules/react-dom/lib/ReactMount.js:320:18)
      at Object._renderSubtreeIntoContainer (node_modules/react-dom/lib/ReactMount.js:401:32)
      at Object.render (node_modules/react-dom/lib/ReactMount.js:422:23)
      at Object.<anonymous> (src/components/App.test.js:7:48)

  ✕ renders without crashing (11ms)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        1.097s
Ran all test suites.
```

### Environment

Run these commands in the project folder and fill in their results:

1. `npm ls react-scripts` (if you haven’t ejected): 

```
auth0-react@0.1.0 /home/christophe/Dev/next/auth0-react
└── (empty)
```

2. `node -v`: v7.6.0
3. `npm -v`: 4.1.2
4. `nvm --version`: 0.32.1

Then, specify:

1. Operating system: Ubuntu 16.04
2. Browser and version: Chrome Version 56.0.2924.87 (64-bit)

### Reproducible Demo

https://github.com/tophsic/auth0-react

**Comment by tophsic** [comment]
Looking at comment https://github.com/facebookincubator/create-react-app/issues/1807#issuecomment-286159704, I tried to downgrade react-router to `^3.0.0'.

I got following error on browser:

```
Error in ./~/auth0-lock/~/auth0-js/src/authentication/index.js
Module not found: 'qs' in /home/christophe/Dev/next/auth0-react/node_modules/auth0-lock/node_modules/auth0-js/src/authentication
```

After `npm install qs`, auth0-react application is operational

I got following error on `npm test` command:

```bash
 FAIL  src/components/App.test.js
  ● Test suite failed to run

    Cannot find module '/home/christophe/Dev/next/auth0-react/node_modules/babel-preset-react-app/node_modules/babel-runtime/regenerator' from 'auth.js'
      
      at Resolver.resolveModule (node_modules/auth0-react-scripts/node_modules/jest/node_modules/jest-cli/node_modules/jest-resolve/build/index.js:144:17)
      at Object.<anonymous> (src/auth.js:1:210)
      at Object.<anonymous> (src/components/App.js:3:39)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.996s
Ran all test suites.
```

Continuing digging

**Comment by tophsic** [comment]
I had errors because I switched to node v4.8.0. With node v7.6.0, tests suite passed with this warning:

```
Warning: [react-router] Location "blank" did not match any routes
```

**Comment by jclancy93** [comment]
The same steps resolved it for me. The issue is that create-react-app now comes with react-router v4 and this project is configured with v3.

**Comment by gaearon** [comment]
Create React App does not come with React Router.

**Comment by dcanning** [comment]
update my package.json to look like this:

{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "auth0-lock": "^10.14.0",
    "dotenv": "^4.0.0",
    "events": "^1.1.1",
    "express": "^4.15.2",
    "express-jwt": "^5.3.0",
    "jwt-decode": "^2.2.0",
    "react": "^15.5.4",
    "react-dom": "^15.5.4",
    "react-router": "^3.0.5"
  },
  "devDependencies": {
    "auth0-react-scripts": "0.6.20"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  },
  "proxy": "http://localhost:3001"
}

then npm i and most was good, app would run, could log in, 2 issues with tests:

  console.error node_modules\react\node_modules\fbjs\lib\warning.js:36
    Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.

  console.error node_modules\warning\warning.js:51
    Warning: [react-router] Location "blank" did not match any routes

**Comment by jDeppen** [comment]
> Warning: Accessing PropTypes via the main React package is deprecated

That was fixed in [v3.0.4](https://github.com/ReactTraining/react-router/releases/tag/v3.0.4)


> Warning: [react-router] Location "blank" did not match any routes

Can anyone offer any tips or point me in the right direction for this one? Thanks
